### ISSUE TITLE
failed to read apk key: open /tmp/apko-3984880358/work/local-melange.rsa.pub: no such file or directory

### ISSUE THREAD

**Issue by rawlingsj** [chainguard-dev/apko#464]
When building locally apko has started to fail this morning.

```
failed to read apk key: open /tmp/apko-3984880358/work/local-melange.rsa.pub: no such file or directory
```

```
--keyring-append /work/local-melange.rsa.pub --repository-append /work/packages
```

full command...
```
docker run --rm -v ${PWD}:/work -w /work cgr.dev/chainguard/apko build latest.apko.yaml foo foo.tar --debug --build-arch amd64 --keyring-append /work/local-melange.rsa.pub --repository-append /work/packages
```

I was able to go back to a previous apko version and things worked again.
```
ghcr.io/chainguard-dev/apko:cf9e6f264bcf38fd8ebbd95fb6a954cce5d16b57
```

**Comment by rawlingsj** [comment]
To recreate you can create a test melange config file `test.melange.yaml`

```yaml
package:
  name: apko-issue-464
  version: 1.2.3
  epoch: 0
  description: Test case for https://github.com/chainguard-dev/apko/issues/464
  target-architecture:
    - all
  copyright:
    - license: Apache-2.0
      paths:
        - "*"

environment:
  contents:
    repositories:
      - https://packages.wolfi.dev/os
    keyring:
      - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
    packages:
      - busybox

pipeline:
  - runs: |
      mkdir -p ${{targets.destdir}}/usr/bin
      echo foo > ${{targets.destdir}}/usr/bin/foo.txt
```

build the apk...
```bash
docker run --platform linux/aarch64 --privileged -v "$PWD":/work cgr.dev/chainguard/melange build /work/test.melange.yaml --arch aarch64 --keyring-append local-melange.rsa.pub --signing-key local-melange.rsa --keyring-append https://packages.wolfi.dev/os/wolfi-signing.rsa.pub --repository-append https://packages.wolfi.dev/os --
```

now create an apko yaml `test.apko.yaml` that references packages in both wolfi and locally
```yaml
contents:
  packages:
    - wolfi-baselayout
    - busybox
    - apko-issue-464

accounts:
  groups:
    - groupname: foo
      gid: 1001
  users:
    - username: foo
      uid: 1001
      gid: 1001
  run-as: root

cmd: sleep infinity

archs:
  - aarch64
```

build apko including the `--keyring-append /work/local-melange.rsa.pub --repository-append /work/packages` flags so we can use the locally built apk
```bash
docker run --rm  --platform linux/aarch64 -v ${PWD}:/work -w /work cgr.dev/chainguard/apko build test.apko.yaml foo foo.tar --debug --build-arch aarch64 --keyring-append /work/local-melange.rsa.pub --repository-append /work/packages --keyring-append https://packages.wolfi.dev/os/wolfi-signing.rsa.pub --repository-append https://packages.wolfi.dev/os
```
Now rerun the same command but using the previous apko image version from a few days ago
swapping
```
cgr.dev/chainguard/apko
```
with
```
ghcr.io/chainguard-dev/apko:cf9e6f264bcf38fd8ebbd95fb6a954cce5d16b57 
```

**Comment by deitch** [comment]
Figured it out. You are correct @rawlingsj it is a regression introduced by #426 

When initializing the keyring, it can read keyfiles from http/https/local files. It assumes that the local files are inside the workdir. In this case, that clearly is not correct, as you meant, "relative to where I called it."

That treatment of "relative to the workdir" is wrong, and I believe I introduced it. It is an easy fix, going to patch it with a PR right away.
